### PR TITLE
Remove snake_case keys from appOptions

### PR DESCRIPTION
**Pull request by joshlory**

### CODE REVIEW

File: dashboard/test/controllers/script_levels_controller_test.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@ class ScriptLevelsControllerTest < ActionController::TestCase
     assert_response :success
     # toolbox blocks comes from project_level not real_level
     app_options = assigns(:level).blockly_options
-    assert_equal '<xml><toolbox/></xml>', app_options[:level]['toolbox_blocks']
+    assert_equal '<xml><toolbox/></xml>', app_options[:level]['toolbox']
```
Comment (Bjvanminnen) [inline dashboard/test/controllers/script_levels_controller_test.rb]: not toolboxBlocks?

File: dashboard/app/models/blockly.rb [hunk 2]
```diff
@@ -165,124 +186,15 @@ def blockly_options
...
-        complete_on_success_condition_not_goals
-      ).map{ |x| x.include?(':') ? x.split(':') : [x,x.camelize(:lower)]}]
-          .each do |dashboard, blockly|
-        # Select value from properties json
-        # Don't override existing valid (non-nil/empty) values
-        property = level_prop[dashboard].presence
-        value = JSONValue.value(level_prop[blockly] || property)
-        level_prop[blockly] = value unless value.nil? # make sure we convert false
-      end
-
```
Comment (Bjvanminnen) [inline dashboard/app/models/blockly.rb]: Removing this block makes me happy :)

File: dashboard/test/controllers/script_levels_controller_test.rb (hunk 1 above)
Comment (joshlory) [inline dashboard/test/controllers/script_levels_controller_test.rb]: Nope, I made that mistake at first (fixed in abb3f90c777f5f31902b70a250551219d784efc7).  The camelization is overridden here: https://github.com/code-dot-org/code-dot-org/pull/2164/files#diff-7c558db0bac67f17752ae9f2a385c24fR158

File: dashboard/app/models/blockly.rb (hunk 2 above)
Comment (joshlory) [inline dashboard/app/models/blockly.rb]: It still sort of exists here, but we don't have to explicitly specify keys anymore: https://github.com/code-dot-org/code-dot-org/pull/2164/files#diff-7c558db0bac67f17752ae9f2a385c24fR166

**Comment by wjordan** [comment]
Very nice! Removing this extra whitelist of properties to convert is going to be SO helpful to prevent common bugs from popping up when that list went out of sync with the level-model properties list. Looks good to me :+1:

File: dashboard/app/models/blockly.rb [hunk 3]
```diff
@@ -149,14 +149,35 @@ def blockly_options
     options = Rails.cache.fetch("#{cache_key}/blockly_level_options") do
       level = self
       # Use values from properties json when available (use String keys instead of Symbols for consistency)
-      level_prop = level.properties.dup || {}
+      level_prop = {}
+
+      # Map Dashboard-style names to Blockly-style names in level object.
+      # Dashboard underscore_names mapped to Blockly lowerCamelCase, or explicit 'Dashboard:Blockly'
```
Comment (bcjordan) [inline dashboard/app/models/blockly.rb]: Should this comment be updated / removed? ` or explicit 'Dashboard:Blockly'` - this portion looks like is no longer the case. maybe "or can be explicitly set in the overrides hash"

File: dashboard/app/models/blockly.rb [hunk 4]
```diff
@@ -149,14 +149,35 @@ def blockly_options
     options = Rails.cache.fetch("#{cache_key}/blockly_level_options") do
       level = self
       # Use values from properties json when available (use String keys instead of Symbols for consistency)
-      level_prop = level.properties.dup || {}
+      level_prop = {}
```
Comment (bcjordan) [inline dashboard/app/models/blockly.rb]: Would now be a good time to rename `level_prop` to something like e.g. `level_props`, `blockly_level` or `blockly_level_object`?

File: dashboard/app/models/blockly.rb [hunk 5]
```diff
@@ -149,14 +149,35 @@ def blockly_options
     options = Rails.cache.fetch("#{cache_key}/blockly_level_options") do
       level = self
       # Use values from properties json when available (use String keys instead of Symbols for consistency)
```
Comment (bcjordan) [inline dashboard/app/models/blockly.rb]: This comment no longer true?

**Comment by bcjordan** [comment]
LGTM, awesome change. Just a couple of notes re: stale comments

Keep an eye on the next DTT if haven't run UI tests with this yet.

**Comment by bcjordan** [comment]
& could be worth quick manual test battery of level editing

**Comment by joshlory** [comment]
Ran UI tests locally and fixed one bug: 8721aeb0c341fad38a2cdcc187030d89a7f14b41.

**Pull request merged**